### PR TITLE
Improved Windows compatibility

### PR DESCRIPTION
**Pull request by arixmkii**
Extracting some work done during experiments implementing https://github.com/containers/podman/issues/13006

First it replaces hardcoded "/dev/null" with os specific variant provided by Go lang. `os.DevNull` was used in other places, so, I assume this is just code cleanup change.

Next it changes use of xzcat to xz as default windows build doesn't have xzcat (in a form of application copy or symlink (not supported on that platform).

From the docs https://linux.die.net/man/1/xzcat "xzcat is equivalent to xz --decompress --stdout." So, it should have no functional changes and preserve the bahavior. Used "-d" and "-c" short keys instead of full ones as there was use of "-k" in short form already.
[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by arixmkii** [comment]
```
If your commit really, truly does not need tests, you can proceed
by adding '[NO NEW TESTS NEEDED]' to the body of your commit message.
```

IMO, it is fine to add this to this PR as the changes are not testable in isolation and should be just covered by regression testing in any form, but will wait for opinions of maintainers.

**Comment by mheon** [comment]
I would consider @giuseppe the closest we have to an SME on compression, so I'll ask him to take a look at the xz/xzcat change; remaining changes LGTM. I think we're good to merge without tests, given everything here should already be exercised by the `podman machine` or E2E tests.

/approve

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
@arixmkii You need to rebase and repush.

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

**Comment by arixmkii** [comment]
Rebased on latest main.

**Comment by mheon** [comment]
One test failure, assuming it's a flake, restarting

/lgtm
/approve
/hold

**Pull request merged**
